Bugfix: Problem compiling MET on MacOS using GCC 12.3.0

## Describe the Problem ##
The compilation of MET-11.1.0-rc1 on the MacOS fails when compiling with gcc 12.3.0, but succeeds with gcc 11.3.0.  The error in the make.log file (attached) is the following when making pb2nc:
[met.make.log](https://github.com/dtcenter/MET/files/12100683/met.make.log)

Note that a libbufr.a file is created from the BUFRLIB compilation, but I noticed this in the log file:

> file: libbufr.a(modv_MXH4WLC.o) has no symbols

[bufr.make.log](https://github.com/dtcenter/MET/files/12100702/bufr.make.log)

```
calcape.f:442:72:

  442 |  510    POLD(KP)=P
      |                                                                        1
Warning: Fortran 2018 deleted feature: DO termination statement which is not END DO or CONTINUE with label 510 at (1)
calcape.f:464:72:

  464 |  530    QSNEW(KP)=QSNEW(KP-1)+DQS
      |                                                                        1
Warning: Fortran 2018 deleted feature: DO termination statement which is not END DO or CONTINUE with label 530 at (1)
calcape.f:472:72:

  472 |  540    PTBL(KP,KTH)=PNEW(KP)
      |                                                                        1
Warning: Fortran 2018 deleted feature: DO termination statement which is not END DO or CONTINUE with label 540 at (1)
calcape.f:485:72:

  485 |  560    THEOLD(KTH)=TH*EXP(ELIWV*QS/(CP*TOLD(KTH)))
      |                                                                        1
Warning: Fortran 2018 deleted feature: DO termination statement which is not END DO or CONTINUE with label 560 at (1)
calcape.f:509:72:

  509 |  580    THENEW(KTH)=THENEW(KTH-1)+DTHE
      |                                                                        1
Warning: Fortran 2018 deleted feature: DO termination statement which is not END DO or CONTINUE with label 580 at (1)
calcape.f:517:72:

  517 |  590    TTBL(KTH,KP)=TNEW(KTH)
      |                                                                        1
Warning: Fortran 2018 deleted feature: DO termination statement which is not END DO or CONTINUE with label 590 at (1)
pb2nc.cc:147:31: error: variable 'const std::array<int, 6> var_gc' has initializer but incomplete type
  147 | const std::array<int, mxr8vt> var_gc = {
      |                               ^~~~~~
pb2nc.cc:151:32: error: aggregate 'std::array<int, 6> bufr_var_code' has incomplete type and cannot be defined
  151 | static std::array<int, mxr8vt> bufr_var_code;
      |                                ^~~~~~~~~~~~~
pb2nc.cc:157:36: error: variable 'const std::array<int, 6> derive_gc' has initializer but incomplete type
  157 | const std::array<int, n_derive_gc> derive_gc = {
      |                                    ^~~~~~~~~
mv -f .deps/pb2nc-pb2nc_conf_info.Tpo .deps/pb2nc-pb2nc_conf_info.Po
make[5]: *** [pb2nc-pb2nc.o] Error 1
```

Another user, @HathewayWill, also tested on a Mac using gcc 12.3.0 and received the same error his met.make.log file.  A copy of his files are linked to in his [comment](https://github.com/dtcenter/MET/issues/2596#issuecomment-1642941330) on another issue.  

### Expected Behavior ###
MET-11.1.0-rc1 should build on the MacOS using gcc 12.3.0.

### Environment ###
Describe your runtime environment:
*1. Machine: Mac Laptop*
*2. OS: MacOS Monterey version 12.6.7*
*3. Software version number: MET-11.1.0-rc1*

### To Reproduce ###
Describe the steps to reproduce the behavior:
Attempt to install MET and its dependent libraries using the compilation script [here](https://raw.githubusercontent.com/dtcenter/MET/bugfix_2596_main_v11.1_rpath_compilation/internal/scripts/installation/compile_MET_all.sh) or after this [PR](https://github.com/dtcenter/MET/pull/2614) is merged, using the script in main_v11.1, on a MacOS using gcc 12.3.0.

### Relevant Deadlines ###
Preferably before the 11.1.0 release.

### Funding Source ###
2792543

## Define the Metadata ##

### Assignee ###
- [ ] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Select **Organization** level **Project** for support of the current coordinated release
- [x] Select **Repository** level **Project** for development toward the next official release or add **alert: NEED CYCLE ASSIGNMENT** label
- [x] Select **Milestone** as the next bugfix version

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Bugfix Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **main_\<Version>**.
Branch name: `bugfix_<Issue Number>_main_<Version>_<Description>`
- [ ] Fix the bug and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **main_\<Version>**.
Pull request: `bugfix <Issue Number> main_<Version> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issue
Select: **Organization** level software support **Project** for the current coordinated release
Select: **Milestone** as the next bugfix version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Complete the steps above to fix the bug on the **develop** branch.
Branch name:  `bugfix_<Issue Number>_develop_<Description>`
Pull request: `bugfix <Issue Number> develop <Description>`
Select: **Reviewer(s)** and **Development** issue
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Close this issue.

--- a/src/tools/other/pb2nc/pb2nc.cc
+++ b/src/tools/other/pb2nc/pb2nc.cc
@@ -54,9 +54,10 @@
 //   015    02/10/18  Halley Gotway  Add message_type_group_map.
 //   016    07/23/18  Halley Gotway  Support masks defined by gen_vx_mask.
 //   017    07/06/22  Howard Soh     METplus-Internal #19 Rename main to met_main
-//   018    09/12/22  Prestopnik     MET #2227 Remove namespace std and netCDF
+//   018    09/12/22  Prestopnik, J. MET #2227 Remove namespace std and netCDF
 //                                   from header files
-//
+//   019    07/21/23  Prestopnik, J. MET #2615 Add #include <array> to compile
+//                                   successfully using gcc12
 ////////////////////////////////////////////////////////////////////////
 
 
@@ -72,6 +73,7 @@
 #include <unistd.h>
 #include <assert.h>
 
+#include <array>
 #include <netcdf>
 
 #include "main.h"

